api/edgeproto/unmarshal_bc.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2024 EdgeXR, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package edgeproto
+
+import (
+	"encoding/json"
+	strings "strings"
+
+	"github.com/edgexr/jsonparser"
+)
+
+// Support for unmarshaling older-object formatted JSON into current
+// objects for backwards compatibility. These functions shall be
+// used at the Master Controller to process data from old
+// CLI/UI clients that are sending the older format JSON.
```

Comment:
Is this here specifically because you don't want to update UI just yet?

Reply:
Exactly :)

Reply:
Ah, I just realized there's a flaw with this - I can deal with the request data, but without a client version, I can't send out the expected response data (like for AppInstShow). I think I will have to fix the UI.